src/views/Articles/ReadArticle/ReadArticle.js

```diff
@@ -0,0 +1,201 @@
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import ReactHtmlParser from 'react-html-parser';
+import { withRouter } from 'react-router-dom';
+import swal from '@sweetalert/with-react';
+import Loader from 'react-loader-spinner';
+import { readArticle, deleteAnArticle } from '@Actions/Articles';
+import connectComponent from '@Lib/connect-component';
+import Icon from '@Components/Icon';
+import { convertToHtml, isEmpty } from '@Utils/';
+import './ReadArticle.scss';
+
+export class ReadArticle extends Component {
+  componentDidMount = async () => {
+    const {
+      getSingleArticle,
+      match: {
+        params: {
+          slug,
+        },
+      },
+    } = this.props;
+    await getSingleArticle(slug);
+  }
+
+  parseArticleBody = article => {
+    if (!article) {
+      return null;
+    }
+    try {
+      return convertToHtml(JSON.parse(article));
+    } catch (e) {
+      return article;
+    }
+  }
+
+  editArticle = () => {
+    const { history, article: { slug } } = this.props;
+    history.push(`/articles/${slug}/edit`);
+  }
+
+  isMyArticle = () => {
+    const {
+      article,
+      auth: {
+        isAuthenticated,
+        user: {
+          id,
+        },
+      },
+    } = this.props;
+    const authorId = !isEmpty(article) ? article.authorId : 0;
+
+    return isAuthenticated && (authorId === id);
+  }
+
+  deleteArticle = () => {
+    swal({
+      text: 'Are you sure? Once deleted, this cannot be undone!',
+      icon: 'warning',
+      buttons: true,
+      dangerMode: true,
+    })
+      .then((willDelete) => {
+        if (willDelete) {
+          const { article: { slug }, history, deleteArticle } = this.props;
+          deleteArticle(slug, history);
+        }
+      });
+  }
+
+  render = () => {
+    const {
+      ui: { loading },
+      article: {
+        title,
+        description,
+        author,
+        createdAt,
+        readTime,
+        image,
+        articleBody,
+        tagList,
+        likesCount,
+        dislikesCount,
+        comment,
+      },
+    } = this.props;
+    const tags = tagList ? tagList.split(' ')
+      .filter(tag => tag.length > 0)
+      .map((tag, i) => (<li key={i}><p>{tag}</p></li>)) : null;
+    const body = this.parseArticleBody(articleBody);
+
+    const loader = (
+      <div data-test="loader" className="loader">
+        <Loader
+          type="ThreeDots"
+          color="#ffa500"
+          height={150}
+          width={150}
+        />
+      </div>
+    );
+
+    return (
+      <>
+        {loading ? loader : (
+          <div className="read-article">
+            <div className="article-header vertical-center">
+              <h1 className="title">{title}</h1>
+              <p className="description">{description}</p>
+              <div className="article-user center">
+                <div className="article-user-details">
+                  <div className="article-img-div center">
+                    <img
```

Comment:
Nice work Chux, I think you should load a custom placeHolder image if no image data comes from the database.
Here is the one we've been using 'https://res.cloudinary.com/drdje1skj/image/upload/v1567427029/profile-placeholder_gvxkia.gif'

Reply:
All right. I will do that now. Thanks